bundles/org.yamcs.studio.core.ui/src/main/java/org/yamcs/studio/core/ui/utils/TimestampFormatter.java

```diff
@@ -0,0 +1,19 @@
+package org.yamcs.studio.core.ui.utils;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+import org.yamcs.studio.core.ui.YamcsUIPlugin;
+
+public class TimestampFormatter {
+
+    public static String format(long instant) {
+        Date date = new Date(instant);
```

Comment:
The incoming long cannot be converted like this to Date. Yamcs instants must be converted via TimeEncoding which is aware of leap seconds. Have a look at the implementation of TimeCatalogue.toString().